src/java.base/share/native/libjli/java.c

```diff
@@ -427,11 +417,16 @@ int
 invokeInstanceMainWithArgs(JNIEnv *env, jclass mainClass, jobjectArray mainArgs) {
     jmethodID constructor = (*env)->GetMethodID(env, mainClass, "<init>", "()V");
     CHECK_EXCEPTION_FAIL();
```

Comment:
Shouldn't this also not be checked until you know you have an instance method?

Reply:
Yes that is what I meant. If the main method is static then the class need not have any such constructor (unless that is part of the spec for this functionality?).